app.ts

```diff
@@ -1,357 +1,62 @@
-        errorResponse
-      );
-    }
-    throw new Error("Something went wrong while generating the pdf");
-  }
-}
-
-async function retrieveReportParts(
-  reportId: string
-): Promise<ReportParts | null> {
-  const reportQuery = `
-  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
-  PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
-  PREFIX besluitvorming: <https://data.vlaanderen.be/ns/besluitvorming#>
-  PREFIX dct: <http://purl.org/dc/terms/>
-  PREFIX prov: <http://www.w3.org/ns/prov#>
-  PREFIX pav: <http://purl.org/pav/>
-
-  SELECT * WHERE {
-    ?s mu:uuid ${sparqlEscapeString(reportId)} .
-    ?s a besluitvorming:Verslag .
- 	  ?piecePart dct:isPartOf ?s .
-    ?piecePart dct:title ?title .
-    ?piecePart prov:value ?htmlContent .
-    FILTER(NOT EXISTS { [] pav:previousVersion ?piecePart }) .
-  }
-  `;
-
-  const {
-    results: { bindings },
-  } = await query(reportQuery);
-  if (bindings.length === 0) {
-    return null;
-  }
-
-  return {
-    annotation: bindings.find(
-      (b: Record<"title", Record<"value", string>>) =>
-        b.title.value === "Annotatie"
-    )?.htmlContent?.value,
-    concerns: bindings.find(
-      (b: Record<"title", Record<"value", string>>) =>
-        b.title.value === "Betreft"
-    ).htmlContent.value,
-    decision: bindings.find(
-      (b: Record<"title", Record<"value", string>>) =>
-        b.title.value === "Beslissing"
-    ).htmlContent.value,
-  };
-}
-
-async function retrieveReportSecretary(
-  reportId: string
-): Promise<Secretary | null> {
-  const dataQuery = `
-    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
-    PREFIX besluitvorming: <https://data.vlaanderen.be/ns/besluitvorming#>
-    PREFIX prov: <http://www.w3.org/ns/prov#>
-    PREFIX dct: <http://purl.org/dc/terms/>
-    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
-    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
-    PREFIX persoon: <https://data.vlaanderen.be/ns/persoon#>
-
-    SELECT DISTINCT ?lastName ?firstName ?title WHERE {
-      ?report mu:uuid ${sparqlEscapeString(reportId)} .
-      ?report a besluitvorming:Verslag .
-      ?report besluitvorming:beschrijft ?decisionActivity .
-      ?decisionActivity prov:wasAssociatedWith ?mandatee .
-      ?mandatee dct:title ?title .
-      ?mandatee mandaat:isBestuurlijkeAliasVan ?person .
-      ?person foaf:familyName ?lastName .
-      ?person persoon:gebruikteVoornaam ?firstName .
-    }
-    `;
-  const queryResult = await query(dataQuery);
-  if (queryResult.results?.bindings?.length) {
-    const result = queryResult.results.bindings[0];
-    return {
-      person: {
-        firstName: result.firstName.value,
-        lastName: result.lastName.value,
-      },
-      title: result.title.value,
-    };
+    const fileMeta = await generateReport(req.params.id, req.headers);
+    res.status(200).send(fileMeta);
+  } catch (e) {
+    res.status(500);
+    console.error(e);
+    res.send(e);
```

Comment:
**issue**: errors "should" be handled using template error handler

See: https://github.com/mu-semtech/mu-javascript-template#error-handling

The service should have a line `app.use(errorHandler); // import { errorHandler } from 'mu'` at the end of the `app.ts` file, when we encounter an error we should then do `next(e)` inside a route handler, this way we ensure that the service sends JSON:API responses (that's the implicit contract of a semantic-works service, that it always communicates using JSON:API).